lib/task/digitalobject/digitalObjectDeleteTask.class.php

```diff
@@ -50,71 +53,132 @@ protected function configure()
    */
   public function execute($arguments = array(), $options = array())
   {
+    $nDeleted = 0;
+    $objectIds = array();
+
+    $this->validMediaTypes = array(
+      'audio' => QubitTerm::AUDIO_ID,
+      'image' => QubitTerm::IMAGE_ID,
+      'text'  => QubitTerm::TEXT_ID,
+      'video' => QubitTerm::VIDEO_ID
+    );
+
     parent::execute($arguments, $options);
 
     $t = new QubitTimer;
-    $id = QubitPdo::fetchColumn('SELECT object_id FROM slug WHERE slug=?', array($arguments['slug']));
 
-    if (!$id)
+    // Remind user they are in dry run mode
+    if ($options['dry-run'])
+    {
+      $this->logSection('digital-object', '*** DRY RUN (no changes will be made to the database) ***');
+    }
+
+    if ($options['media-type'] && !array_key_exists($options['media-type'], $this->validMediaTypes))
+    {
+      error_log(sprintf('Invalid value for "media-type", must be one of (%s)',
+        implode(',', array_keys($this->validMediaTypes))));
+
+      exit(1);
+    }
+
+    $sql = "SELECT slug.object_id, object.class_name \r
+              FROM slug \r
+              JOIN object ON object.id = slug.object_id \r
+              WHERE slug.slug = '" . $arguments['slug'] . "'";
```

Comment:
Nitpick: the `\r` are not really needed.

Reply:
Removed.